src/liquid/utils/cloneAttributes.ts

```diff
@@ -1,28 +1,27 @@
-import { JSXBase } from '@stencil/core/internal'
-
-export const cloneAttributes = <B extends HTMLElement>(
-  el: B,
-  attributesToIgnore: string[] = []
-) =>
-  Object.values(el.attributes).reduce<JSXBase.HTMLAttributes<B>>(
-    (acc, { name, value }) => {
-      if (
-        !name ||
-        [
-          'style',
-          'id',
-          'class',
-          'slot',
-          'part',
-          ...attributesToIgnore,
-        ].includes(name)
-      ) {
-        return acc
-      }
+export function cloneAttributes(attributesToIgnore: string[] = []) {
+  const attributes = []
+  for (const key in this) {
+    // Component props are getters. Getters don't have a descriptor.
+    // So we can check for component props as follows:
+    if (!Object.getOwnPropertyDescriptor(this, key) && this[key]) {
```

Comment:
Not sure if this `&& this[key]` is necessary here...? Also, don't we also want to clone `prop={false}` values?

```suggestion
    if (!Object.getOwnPropertyDescriptor(this, key)) {
```

Reply:
Good catch. This is now changed.